data/src/main/java/org/apache/iceberg/data/parquet/GenericParquetWriter.java

```diff
@@ -284,6 +290,11 @@ public void write(int repetitionLevel, byte[] value) {
     }
   }
 
+  public interface StructWriterFactory {
+
+    StructWriter<?> create(List<ParquetValueWriter<?>> writers);
```

Comment:
I tried to mark this method to be `protected` but seems java8 don't allow to do that....

Reply:
Interface methods are always public.